### PR TITLE
Edit task UI

### LINKED ISSUE #56: Edit task details
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Feature

**Edit task details**
It should be possible to edit the task details.

**Describe the solution you'd like**
The edit should take place within a popover form. The popover should be shown if the user either double clicks on a task row, or selects the task row and hits the Enter button. Similar to what Apple Reminders does:

Bildschirmfoto 2019-10-22 um 19 50 10

### PR DESCRIPTION
**Pull request by marbetschar**
This fixes #56 and adds the minimal features of the task settings popover. Here's how it looks like right now:

Screenshot von 2019-11-08 14 14 03

### CODE REVIEW

**Comment by alainm23** [comment]
I like the design, but I feel that the Gtk.Switch is very big, maybe we could make it smaller, try with: 

```
.switch slider {
    min-height: 16px;
    min-width: 16px;
}
```

**Comment by marbetschar** [comment]
Agreed, it is. But I feel the same way about the Gtk.Switch in the list settings popover - and therefore preferred to stick to this default size. If, then it should either be made smaller on a system- or at least app wide level.

Gtk-Switch

**Comment by marbetschar** [comment]
Feedback from @tintou:

Replace `TaskModel` with `ECal.Component`: Create it manually on `objects_added`. [See elementary/Calendar for an example](https://github.com/elementary/calendar/blob/9bf852c798033122d3e8a978d09244f608a296b3/core/Model/CalendarModel.vala#L398-L409).

There's a lot of `widget.margin_start/end/top/bottom` usually if you use a `Gtk.Grid` you can use `row_spacing` and `column_spacing` to not have this.

**Comment by marbetschar** [comment]
@tintou I tried to replace the `TaskModel` with the `ECal.Component` - but didn't have too much luck with it:

- I'm not able to retrieve the due date correctly. I assume the [libecal-vapi for `ECal.ComponentDateTime` is invalid (maybe due to version mismatch)](https://github.com/elementary/tasks/blob/50cdb168f20f6ddd67f1519f7d567ab51d891a6a/vapi/libecal-1.2.vapi#L311-L314) - at least it does not reflect what's on [valadoc.org](https://valadoc.org/libecal-2.0/ECal.ComponentDateTime.html)
- The task summary in the popover outputs total gibberish
- The output in the popover sometimes lacks the description - for whatever reason this does not always happen

See these screenshots for further info on the above points:

Screenshot von 2019-11-12 15 16 20
Screenshot von 2019-11-12 15 14 15

**Comment by marbetschar** [comment]
@tintou or @danrabbit  whats missing to get this merged?

**Comment by tintou** [comment]
Ideally it would be great to fix the above comments but that's just neat-pick things ;)

File: src/Widgets/TaskSettingsPopover.vala [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        grid.add (new Gtk.Separator (Gtk.Orientation.HORIZONTAL));
+        grid.add (due_button);
+        grid.add (due_datetimepicker);
+        grid.add (new Gtk.Separator (Gtk.Orientation.HORIZONTAL));
+        grid.add (description_scrolled_window);
+        grid.add (new Gtk.Separator (Gtk.Orientation.HORIZONTAL));
+        grid.add (done_button);
+        grid.show_all ();
+
+        width_request = 379;
```
[the remark this answers is not in the document]
Comment (marbetschar) [reply src/Widgets/TaskSettingsPopover.vala]: @tintou yeah 😅
not quite sure if 379 is the way to go - but it seems to work for now ;)

**Pull request merged**